android/build.gradle

```diff
@@ -43,11 +43,11 @@ android {
      *                   on Android.
      *
      * are defined in gradle.properties file. This file is
-     * updated by QtCreator and androiddeployqt tools.
+     * updated by QtCredftor and androiddeployqt tools.
      * Changing them manually might break the compilation!
      *******************************************************/
 
-    compileSdkVersion androidCompileSdkVersion.toInteger()
+    compileSdkVersion 29
```

Comment:
I'm confused about this. The comments here say, the `compileSdkVersion` should be defined in `gradle.properties`. This file does not exist. If I create it with the content `androidCompileSdkVersion=29` it makes no difference. 
`compileSdkVersion` is `android-21` because of `androidCompileSdkVersion.toInteger()` what returns `21` - no idea where it gets this information from.

With defining `compileSdkVersion 29` everything works fine but I don't think it's the proper way.

Reply:
It is created when you cross build. In `build/android-build`

Reply:
Also see https://github.com/opengisch/OSGeo4A/blob/master/.docker/qt-ndk/Dockerfile#L5

Reply:
So would it do it with `android-21` -> `android-29`

Reply:
There are other definitions in `src/apk/project.properties` and the `tools/android.toolchain.cmake.r19b` is setting it as well to `android-21` and in `config.conf.default`. Just to know. I can do some research and tries but I believe, that you @m-kuhn can give me a lead where it's important to change and why...